tests/test_deprecated.py

```diff
@@ -718,49 +718,51 @@ class Model(BaseModel):
 
     assert hasattr(parse_obj_as, '__deprecated__')
     with pytest.warns(
-        PydanticDeprecatedSince20, match='parse_obj_as is deprecated. Use pydantic.TypeAdapter.validate_python instead.'
+        PydanticDeprecatedSince20,
+        match='`parse_obj_as` is deprecated. Use `pydantic.TypeAdapter.validate_python` instead.',
     ):
         parse_obj_as(Model, {'x': 1})
 
     assert hasattr(schema_json_of, '__deprecated__')
     with pytest.warns(
-        PydanticDeprecatedSince20, match='schema_json_of is deprecated. Use pydantic.TypeAdapter.json_schema instead.'
+        PydanticDeprecatedSince20,
+        match='`schema_json_of` is deprecated. Use `pydantic.TypeAdapter.json_schema` instead.',
     ):
         schema_json_of(Model)
 
     assert hasattr(schema_of, '__deprecated__')
     with pytest.warns(
-        PydanticDeprecatedSince20, match='schema_of is deprecated. Use pydantic.TypeAdapter.json_schema instead.'
+        PydanticDeprecatedSince20, match='`schema_of` is deprecated. Use `pydantic.TypeAdapter.json_schema` instead.'
     ):
         schema_of(Model)
 
     assert hasattr(load_str_bytes, '__deprecated__')
-    with pytest.warns(PydanticDeprecatedSince20, match='load_str_bytes is deprecated.'):
+    with pytest.warns(PydanticDeprecatedSince20, match='`load_str_bytes` is deprecated.'):
         load_str_bytes('{"x": 1}')
 
     assert hasattr(load_file, '__deprecated__')
     file = tmp_path / 'main.py'
     file.write_text('{"x": 1}')
-    with pytest.warns(PydanticDeprecatedSince20, match='load_file is deprecated.'):
+    with pytest.warns(PydanticDeprecatedSince20, match='`load_file` is deprecated.'):
         load_file(file)
 
     assert hasattr(pydantic_encoder, '__deprecated__')
     with pytest.warns(
-        PydanticDeprecatedSince20, match='pydantic_encoder is deprecated, use BaseModel.model_dump instead.'
+        PydanticDeprecatedSince20,
+        match='`pydantic_encoder` is deprecated, use `pydantic_core.to_jsonable_python` instead.',
     ):
         pydantic_encoder(Model(x=1))
 
     assert hasattr(custom_pydantic_encoder, '__deprecated__')
     with pytest.warns(
-        PydanticDeprecatedSince20, match='custom_pydantic_encoder is deprecated, use BaseModel.model_dump instead.'
+        PydanticDeprecatedSince20, match='`custom_pydantic_encoder` is deprecated, use `BaseModel.model_dump` instead.'
     ):
         custom_pydantic_encoder({int: lambda x: str(x)}, Model(x=1))
 
     assert hasattr(timedelta_isoformat, '__deprecated__')
     with pytest.warns(PydanticDeprecatedSince20, match='timedelta_isoformat is deprecated.'):
         timedelta_isoformat(timedelta(seconds=1))
 
-    assert all(hasattr(func, '__deprecated__') for func in get_overloads(validate_arguments))
```

Comment:
For this assertion to be true, each decorator would have to be marked as deprecated, which is unnecessary. See https://discuss.python.org/t/how-should-warnings-deprecated-be-handled-with-typing-overload/40641 for more details.